lib/run_loop/device_agent/client.rb

```diff
@@ -574,6 +574,45 @@ def springboard_alert_visible?
       end
 
       # @!visibility private
+      def dismiss_springboard_alert(button_title)
+        parameters = { :button_title => button_title }
+        request = request("dismiss-springboard-alert", parameters)
+        client = http_client(http_options)
+        response = client.post(request)
+        hash = expect_300_response(response)
+
+        if hash["error"]
+          raise RuntimeError, %Q[
+Could not dismiss SpringBoard alert by touching button with title '#{button_title}':
+
+#{hash["error"]}
+
+]
+        end
+        true
+      end
+
+      # @!visibility private
+      def dismiss_springboard_alerts_automatically
```

Comment:
I don't have a long history with ruby idioms, but with most languages in this case I'd expect an API to accept a parameter instead of have the parameter implicitly dictated by the method name:

```c
dismiss_springboard_alerts_automatically(true);
```
```java
dismissSpringboardAlertsAutomatically(true);
```
```objc
[self dismissSpringboardAlertsAutomatically:YES];
```

If you wanted to force more semantic clarity on the caller, maybe something like
```ruby
set_springboard_alert_dismissal(SpringboardDismissal::MANUAL)
```

In my mind, if I were programming and I realized I wanted to change the state of springboard automatic dismissal, I'd assume I'm going to be _set_ting a state and thus would instinctively look for a method whose name was similar to `set_<something>` or otherwise more clearly implied a change of state. 

If I'm not mistaken, ruby semantics would suggest at the very least that the method names, since they are changing state/have side effects, be `dismiss_springboard_alerts_automatically!` and `dismiss_springboard_alerts_manually!`, no?

A more clear approach in my opinion might be 
```ruby
enable_automatic_springboard_alert_dismissal!
disable_automatic_springboard_alert_dismissal!
```

Also, I think these two methods could be refactored - seems they only have a single variable differentiating their implementations.